x/protorev/keeper/posthandler.go

```diff
@@ -128,7 +128,13 @@ func (k Keeper) ProtoRevTrade(ctx sdk.Context, swappedPools []SwapToBackrun) (er
 
 		// The error that returns here is particularly focused on the minting/burning of coins, and the execution of the MultiHopSwapExactAmountIn.
 		if maxProfitAmount.GT(sdk.ZeroInt()) {
-			if err := k.ExecuteTrade(ctx, optimalRoute, maxProfitInputCoin); err != nil {
+			// Create backrun event to be emitted if the trade is executed successfully
+			backrunEvent, err := k.CreateBackrunEvent(ctx, pool, remainingPoolPoints)
+			if err != nil {
+				return err
+			}
```

Comment:
cc: the comment about merging the creation and emission of event into a single function.

Why are we building the event here and appending more attributes later instead of just building the entire event in one place?

Reply:
imo I would just pass in the parameters `pool` and `remainingPoolPoints` into execute trade.

Reply:
Yeah I agree, the reason I did it this way was that I only had to pass in the event as a param to append to, but I like the idea of making one function that does it all in one place.

Reply:
combined here: 29f182574e40d3eb59d71047f59ad883835d8ed3